### PR TITLE
feat: add EthereumEipXXXXTransaction support

### LINKED ISSUE #1742: Implement native Ethereum transaction data construction and signing
## Summary

The Hiero Go SDK can serialize the four Ethereum transaction data types (Legacy, EIP-2930, EIP-1559, EIP-7702) but cannot **sign** them natively — a developer must compute the secp256k1 signature, recovery id, and RLP layout themselves (typically via an external library) before submitting through `EthereumTransaction`. On `main`, every field is stored as raw `[]byte`, the access list is a `[][]byte`, and there are no typed accessors. This issue tracks bringing the Go SDK to parity with the cross-SDK [Native Ethereum Transaction Data design proposal](https://github.com/hiero-ledger/sdk-collaboration-hub/pull/266): a uniform `Sign(key)`, a structured `AccessListItem`, additive typed accessors layered over the existing fields, the ergonomic `EthereumTransaction.SetEthereumData(data)` overload, and the `EthereumTransactionData` naming alignment.

The implementation should follow the [SDK design document](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/proposals/native-ethereum-transaction-data.md) and the patterns of the existing `sdk/ethereum_eip1559_transaction.go` and the other transaction-data files.

## Problem

Today, to submit an `EthereumTransaction` a developer must hand-build the raw RLP, prepend the correct type prefix (`0x01`/`0x02`/`0x04`), compute the secp256k1 recovery id, sign the payload, split `r`/`s`, and re-encode. This is error-prone (RLP big-integer trimming, type-prefix placement, recovery-id computation) and inconsistent across SDKs.

Current state on `main` — all four transaction-data types exist, but the supporting ergonomics do not:

- **No native `Sign`** on any type (the universal gap across all SDKs).
- **Fields are stored as raw exported `[]byte`** (e.g. `Nonce`, `Value`, `ChainId`) with **no typed accessors** — no getters/setters at all.
- **The access list is a `[][]byte`**, not an inspectable, structured type.
- **The EIP-7702 authorization is a positional `AuthorizationTuple [6][]byte`** (`[chainId, contractAddress, nonce, yParity, r, s]`), not the named, structured `Authorization` the proposal adopts as canonical.
- **No `EthereumTransaction.SetEthereumData(data)` overload** accepting a data object directly — callers must manually call `ToBytes`/`Sign` and thread the byte array.
- **Naming collision:** Go's `EthereumTransactionData` is a concrete *union wrapper* (it holds one of the four variants), whereas in JS/Java the same name is the *abstract base type*. The proposal adopts the abstract-base meaning as canonical and recommends Go rename its wrapper.

### Native signing

- Add `Sign(key PrivateKey) ([]byte, error)` to all four types:
  - `EthereumLegacyTransaction`
  - `EthereumEIP2930Transaction`
  - `EthereumEIP1559Transaction`
  - `EthereumEIP7702Transaction`
- `Sign` ECDSA-signs the unsigned RLP payload (type-prefixed message for typed variants), writes `r`, `s`, and the recovery component (`recoveryId` for typed, `v` for legacy) back onto the receiver, and returns the signed RLP bytes ready for `SetEthereumData`.
- Signing with a non-ECDSA key (e.g. Ed25519) **must fail** with a clear error. Detection is via the recovery-id computation, not a key-type or length check (an Ed25519 signature is also 64 bytes).
- Factor the shared sign/encode logic into common helpers reused by the three typed variants rather than duplicating it per variant.

### Structured types (additive, no breaking changes)

The existing public byte-backed fields must stay exactly as they are — the structured types are layered **on top** as new accessors that read from / serialize into those same fields. Nothing existing is removed or retyped.

- **`AccessListItem`** — introduce a named, structured type (`Address []byte`, `StorageKeys [][]byte`, `AddStorageKey(key)`). **Keep the existing public `AccessList [][]byte` field unchanged.** Add `SetAccessListItems([]AccessListItem)` / `AddAccessListItem(...)` setters that serialize the structured items **into** the existing `AccessList [][]byte`, and a `GetAccessListItems()` getter that parses them back out. The byte field stays the source of truth; the struct is just a convenient view.
- **`Authorization`** (HIP-1340) — add the same additive treatment for the EIP-7702 authorization: keep the existing `AuthorizationList []AuthorizationTuple` field, and add a named/structured `Authorization` (`chainId`, `address`, `nonce`, `yParity`, `r`, `s`) with `AddAuthorization(...)` / structured getter/setter that read/write the existing field. Deprecate the positional tuple rather than removing it.

### Additive typed accessors

- Add typed + raw-bytes accessor **pairs** for every field, using Go's `…Bytes` suffix convention (Go has no overloading): `GetValue()`/`GetValueBytes()` and `SetValue(...)`/`SetValueBytes(...)`. Typed forms: `uint64` for `chainId`/`nonce`/`gasLimit`; `*big.Int` for `gasPrice`/`maxPriorityGas`/`maxGas`/`value`; `[]byte` for `to`/`callData`/`r`/`s`. Setters return the concrete pointer for chaining.
- Keep the existing exported `[]byte` fields for backwards compatibility — accessors are **additive**, nothing is removed or retyped.

### Updated API

- Add an `EthereumTransaction.SetEthereumData(data *EthereumTransactionData)` (data-object-accepting) overload. When passed unsigned data (no `r`/`s`), it must either serialize the already-signed data or return a clear error — it must **never** silently submit an unsigned transaction. The existing `SetEthereumData([]byte)` semantics must not change.

### Naming alignment (non-blocking)

- Introduce a shared `EthereumTransactionBody` interface (`ToBytes`, `Sign`, `String`) implemented by all four variants, and rename the Go union wrapper so `EthereumTransactionData` consistently denotes the abstract base across SDKs. This is a naming change only — the field/behavior contract is unaffected. Deprecate, don't remove, per repo convention.

### Patterns to follow

`sdk/ethereum_eip1559_transaction.go` is the primary reference. Each variant should:

- Split the existing `ToBytes()` into an unsigned-RLP builder + a sign-and-encode path so `Sign` and `ToBytes` share one canonical encoding (a signed and a re-serialized transaction can never drift).
- Use shared RLP integer helpers (minimal big-endian: leading zero bytes trimmed; zero encoded as empty bytes) for the typed accessors, so every field encodes identically.

### Key files

**Modify** (under `sdk/`):
- `ethereum_legacy_transaction.go`, `ethereum_eip2930_transaction.go`, `ethereum_eip1559_transaction.go`, `ethereum_eip7702_transaction.go` — add `Sign`, split `ToBytes`, add typed accessors
- `ethereum_transaction_data.go` — add the `EthereumTransactionBody` interface, wrapper delegation, shared `Sign`/encode helpers; naming alignment
- `ethereum_transaction.go` — `SetEthereumData(data)` overload

**Add** (under `sdk/`):
- `access_list_item.go` — the structured `AccessListItem` type + RLP round-trip helpers
- a shared encoding helpers file (e.g. `ethereum_encoding.go`) for canonical `uint64`/`*big.Int` ⇄ bytes

### Testing strategy

Implement the 14 Given/When/Then cases from the [design doc Test Plan](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/proposals/native-ethereum-transaction-data.md#test-plan):

1. **Unit tests** (`*_unit_test.go`, build tag `unit`): per-variant `Sign` populates the signature fields and `ToBytes` carries the correct prefix; `FromBytes`/`ToBytes` round-trip equality; access-list and authorization-list preservation (incl. empty-not-null); large 256-bit values and zero values encode with minimal big-endian; non-ECDSA key fails.
2. **Integration tests** (`*_e2e_test.go`, build tag `e2e`): build → sign → `SetEthereumData` → execute against a test network for a signed EIP-1559 contract call (case 12) and an EIP-7702 delegation (case 13); unsigned-data overload behavior (case 14).

## Acceptance Criteria

- [ ] `Sign(key)` implemented on all four variants, with shared sign/encode helpers (no per-variant duplication)
- [ ] Signing with a non-ECDSA key fails with a clear error (detected via recovery-id computation)
- [ ] `Sign` and `ToBytes` share one canonical RLP encoding path
- [ ] Structured `AccessListItem` added with getter/setter that read/write the existing `AccessList [][]byte` field — the public byte field is kept, not replaced or retyped
- [ ] EIP-7702 authorization gets a named/structured `Authorization` view over the existing `AuthorizationList` field (old `AuthorizationTuple` deprecated, not removed)
- [ ] Typed + `…Bytes` accessor pairs added for every field; existing exported `[]byte` fields kept (additive)
- [ ] `EthereumTransaction.SetEthereumData(data)` overload added; never submits an unsigned transaction; existing `SetEthereumData([]byte)` unchanged
- [ ] `EthereumTransactionBody` interface added and `EthereumTransactionData` naming aligned with the abstract-base meaning (old name deprecated, not removed)
- [ ] Unit tests cover the design-doc test plan (sign, round-trip, access/authorization lists, large/zero values, non-ECDSA failure)
- [ ] Integration tests cover the signed EIP-1559 and EIP-7702 end-to-end flows
- [ ] Backwards compatibility maintained; all CI checks pass

## References

- **Design proposal (PR):** https://github.com/hiero-ledger/sdk-collaboration-hub/pull/266
- **SDK design document:** https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/proposals/native-ethereum-transaction-data.md
- **HIP-1340 — EOA Code Delegation (EIP-7702):** https://hips.hedera.com/hip/hip-1340
- **EIPs:** [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718), [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930), [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559), [EIP-7702](https://eips.ethereum.org/EIPS/eip-7702)

### PR DESCRIPTION
**Pull request by Dosik13**
**Description**:

  Add native construction and signing of Ethereum transaction data for all four formats (Legacy, EIP-2930, EIP-1559, EIP-7702), so a developer can build and sign an Ethereum transaction inside the SDK and submit it through the existing EthereumTransaction. Implements the cross-SDK [proposal in sdk-collaboration-hub.](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/proposals/native-ethereum-transaction-data.md)

  - Add EthereumTransactionBody interface (Sign, ToBytes, String) implemented by the Legacy, EIP-2930, EIP-1559, and EIP-7702 transaction types
  - Add native Sign(key) to each transaction type — secp256k1 signing with recovery-id computation, writing R/S and V/RecoveryId back onto the receiver and returning the signed RLP; reject non-ECDSA keys
  - Extract shared _signTypedTransaction and _encodeTypedWithSignature helpers for the typed variants, and split serialization into _toUnsignedRLP / _encodeWithSignature
  - Add paired typed and raw-bytes accessors for every field across all four types (GetValue/SetValue and GetValueBytes/SetValueBytes)
  - Add Ethereum RLP canonical integer encoding helpers (uint64/big.Int to minimal big-endian, zero as empty bytes)
  - Add structured AccessListItem type with getters/setters, AddStorageKey, and RLP helpers; expose it through GetAccessListItems/SetAccessListItems/AddAccessListItem
  - Replace the positional AuthorizationTuple [6][]byte with a structured Authorization type (chainId, address, nonce, yParity, r, s) with constructor, accessors, String, and RLP helpers
  - Add EthereumTransaction.SetEthereumDataFromBody(body) to serialize a signed transaction body directly, returning an error rather than submit an unsigned transaction
  - Add EthereumTransactionData wrapper dispatch over the four types (FromBytes, ToBytes, Sign, GetTransaction, GetData/SetData)
  - Add unit tests covering signing, RLP round-trips, accessors, access lists, authorization lists, and decode error paths

**Related issue(s)**:

Fixes #1742

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/access_list_item.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+			storageKeys = append(storageKeys, keyItem.itemValue)
+		}
+	}
+
+	return NewAccessListItem(address, storageKeys), nil
+}
+
+// _accessListItemsFromBytes converts [][]byte into a slice of AccessListItem.
+// Entries that fail to decode are skipped.
+func _accessListItemsFromBytes(data [][]byte) []AccessListItem {
```
Comment (0xivanov) [inline sdk/access_list_item.go:112]: 🚀

**Review by gsstoykov — APPROVED** [review]
